### PR TITLE
validation of parameters with the same name

### PR DESCRIPTION
**Pull request by willyborankin**

### CODE REVIEW

File: src/test/java/io/aiven/kafka/connect/gcs/config/GcsSinkConfigTest.java [hunk 1]
```diff
@@ -86,6 +88,26 @@ final void incorrectFilenameTemplatesForTopicPartitionRecord(final String templa
...
+                "gcs.bucket.name", "asdasd"
+            );
+        final Template t = new GcsSinkConfig(properties).getFilenameTemplate();
+        final String fileName = t.instance()
+            .bindVariable("topic", () -> "a")
+            .bindVariable("timestamp", () -> "t")
+            .bindVariable("partition", () -> "p")
+            .bindVariable("start_offset", () -> "c")
+            .render();
+        assertEquals("a-t-t-t-p-c.gz", fileName);
```
Comment (ivanyu) [inline src/test/java/io/aiven/kafka/connect/gcs/config/GcsSinkConfigTest.java:108]: A proposal for a little more strict checking:
```suggestion
            .bindVariable("timestamp", VariableTemplatePart.Parameter::value)
            .bindVariable("partition", () -> "p")
            .bindVariable("start_offset", () -> "c")
            .render();
        assertEquals("a-YYYY-MM-dd-p-c.gz", fileName);
```

Comment (willyborankin) [reply src/test/java/io/aiven/kafka/connect/gcs/config/GcsSinkConfigTest.java:108]: then for `start_offset` same thing i will add it

**Comment by ivanyu** [comment]
We don't have an integration test for timestamps. Probably, it's worth adding.

File: src/test/java/io/aiven/kafka/connect/gcs/config/GcsSinkConfigTest.java (hunk 1 above)
Comment (willyborankin) [reply src/test/java/io/aiven/kafka/connect/gcs/config/GcsSinkConfigTest.java]: fixed

**Pull request merged**